corehq/apps/geospatial/reports.py

```diff
@@ -42,22 +45,48 @@ def template_context(self):
     @property
     def report_context(self):
         cases = []
+        invalid_geo_cases_count = 0
         for row in self.es_results['hits'].get('hits', []):
             es_case = self.get_case(row)
             display = CaseDisplayES(es_case, self.timezone, self.individual)
+
+            coordinates = _get_geo_location(es_case)
+            if coordinates is None:
+                invalid_geo_cases_count += 1
+                continue
             # We should consider passing in a "center_coordinates" fields to center the map
             # to the relavent
             case = {
                 "case_id": display.case_id,
                 "case_type": display.case_type,
                 "name": display.case_name,
-                "coordinates": _get_geo_location(es_case)
+                "coordinates": coordinates
             }
             cases.append(case)
+
+        invalid_cases_link = self._invalid_geo_cases_report_link if invalid_geo_cases_count else ''
+
         return dict(
-            context={"cases": cases},
+            context={
+                "cases": cases,
+                "invalid_geo_cases_report_link": invalid_cases_link,
+            },
         )
 
     @property
     def default_report_url(self):
         return reverse('geospatial_default', args=[self.request.project.name])
+
+    @property
+    def _invalid_geo_cases_report_link(self):
+        cle = CaseListExplorer(self.request, domain=self.domain)
+        query_params = "search_xpath={search_xpath}".format(
```

Comment:
Currently adding the filters to the URL is being done manually. Not sure if there's a better way of doing it in our codebase?

Reply:
You could use Django's `QueryDict` class to construct the query params. This will look something like the following:
```
from django.http import QueryDict
url = 'https://foobar.com/'
q = QueryDict()
q['search_xpath'] = quote(f"{GEO_POINT_CASE_PROPERTY} = ''")
if self.case_type:
    q['case_type'] = self.case_type
return "{url}?{query_string}".format(url=url, query_string=q.urlencode())
```